accountsdb-plugin-postgres/src/postgres_client/postgres_client_transaction.rs

```diff
@@ -0,0 +1,1271 @@
+        Self {
+            index: instructions.index as i16,
+            instructions: instructions
+                .instructions
+                .iter()
+                .map(DbCompiledInstruction::from)
+                .collect(),
+        }
+    }
+}
+
+fn get_reward_type(reward: &Option<RewardType>) -> Option<String> {
+    reward.as_ref().map(|reward_type| match reward_type {
+        RewardType::Fee => "fee".to_string(),
+        RewardType::Rent => "rent".to_string(),
+        RewardType::Staking => "staking".to_string(),
+        RewardType::Voting => "voting".to_string(),
+    })
+}
+
+impl From<&Reward> for DbReward {
+    fn from(reward: &Reward) -> Self {
+        Self {
+            pubkey: reward.pubkey.clone(),
+            lamports: reward.lamports as i64,
+            post_balance: reward.post_balance as i64,
+            reward_type: get_reward_type(&reward.reward_type),
+            commission: reward
+                .commission
+                .as_ref()
+                .map(|commission| *commission as i16),
+        }
+    }
+}
+
+fn get_transaction_status(result: &Result<(), TransactionError>) -> Option<String> {
+    if result.is_ok() {
+        return None;
+    }
+
+    let err = match result.as_ref().err().unwrap() {
+        TransactionError::AccountInUse => "AccountInUse",
+        TransactionError::AccountLoadedTwice => "AccountLoadedTwice",
+        TransactionError::AccountNotFound => "AccountNotFound",
+        TransactionError::ProgramAccountNotFound => "ProgramAccountNotFound",
+        TransactionError::InsufficientFundsForFee => "InsufficientFundsForFee",
+        TransactionError::InvalidAccountForFee => "InvalidAccountForFee",
+        TransactionError::AlreadyProcessed => "AlreadyProcessed",
+        TransactionError::BlockhashNotFound => "BlockhashNotFound",
+        TransactionError::InstructionError(idx, error) => {
+            return Some(format!(
+                "InstructionError: idx ({}), error: ({})",
+                idx, error
+            ));
+        }
+        TransactionError::CallChainTooDeep => "CallChainTooDeep",
+        TransactionError::MissingSignatureForFee => "MissingSignatureForFee",
+        TransactionError::InvalidAccountIndex => "InvalidAccountIndex",
+        TransactionError::SignatureFailure => "SignatureFailure",
+        TransactionError::InvalidProgramForExecution => "InvalidProgramForExecution",
+        TransactionError::SanitizeFailure => "SanitizeFailure",
+        TransactionError::ClusterMaintenance => "ClusterMaintenance",
+        TransactionError::AccountBorrowOutstanding => "AccountBorrowOutstanding",
+        TransactionError::WouldExceedMaxAccountCostLimit => "WouldExceedMaxAccountCostLimit",
+        TransactionError::WouldExceedMaxBlockCostLimit => "WouldExceedMaxBlockCostLimit",
+        TransactionError::UnsupportedVersion => "UnsupportedVersion",
+        TransactionError::InvalidWritableAccount => "InvalidWritableAccount",
```

Comment:
Can we just use 

```
 result.as_ref().err().unwrap().to_string()
```

`TransactionError` uses the `error` messages on its fields: https://github.com/solana-labs/solana/blob/cb0bb5bd1e90e87ff04fa10e9a66e48f920f5472/sdk/src/transaction/mod.rs#L40, which should automatically generate a `Display` attribute according to https://docs.rs/thiserror/1.0.30/thiserror/, and `ToString` should be implemented for any `Display` types https://doc.rust-lang.org/std/string/trait.ToString.html

Reply:
On second thought, this might not be so great if we ever change those attributes on the `TransactionError`, would cause a subtle compatibility issue

Reply:
What do you think if instead of returning a string we make a custom `DbTransactionError` type that implements `From<TransactionError>`. We can also mark this enum with the `frozen_abi` tag like https://github.com/solana-labs/solana/blob/cb0bb5bd1e90e87ff04fa10e9a66e48f920f5472/core/src/consensus.rs#L105 to catch breaking changes

Reply:
Sure -- let me look into it.

Reply:
I chose to use Postgres enum types to make the types stronger for transaction error and reward types.